src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -2242,26 +2253,29 @@
          job_id text,
          report_type text DEFAULT 'agent' NOT NULL) PARTITION BY RANGE (%s)" date-column))
 
-    ; Moving existing children to partitioned table
+    ;; Moving existing children to partitioned table
     (reattach-partitions table child-table-maps)
 
-    ; Drop redundant child constraints
+    ;; Drop redundant child constraints
     (drop-child-partition-constraints date-column child-table-maps)
 
+    ;; Drop existing foreign keys from partitions
+    (drop-foreign-key-constraints child-table-maps)
```

Comment:
Just occurred to me that while it's true this migration isn't released externally yet, this will, I think, break z to z upgrades internally (pe-next and the like).